src/Microsoft.Data.SqlClient/tests/Common/LocalAppContextSwitchesHelper.cs

```diff
@@ -0,0 +1,355 @@
+// the switches and restores them when disposed.
+//
+// This follows the RAII pattern to ensure that the switches are always
+// restored, which is important for global state like LocalAppContextSwitches.
+//
+// https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization
+//
+public sealed class LocalAppContextSwitchesHelper : IDisposable
+{
+    #region Public Types
+
+    // This enum is used to represent the state of a switch.
+    //
+    // It is a copy of the Tristate enum from LocalAppContextSwitches.
+    //
+    public enum Tristate : byte
+    {
+        NotInitialized = 0,
+        False = 1,
+        True = 2
+    }
+
+    #endregion
+
+    #region Construction
+
+    // Construct to capture all existing switch values.
+    //
+    // Fails the test if any values cannot be captured.
+    //
+    public LocalAppContextSwitchesHelper()
+    {
+        // Acquire a handle to the LocalAppContextSwitches type.
+        var assembly = typeof(SqlCommandBuilder).Assembly;
+        var switchesType = assembly.GetType(
+            "Microsoft.Data.SqlClient.LocalAppContextSwitches");
+        if (switchesType == null)
+        {
+            Assert.Fail("Unable to find LocalAppContextSwitches type.");
+        }
+
+        // A local helper to acquire a handle to a property.
+        void InitProperty(string name, out PropertyInfo property)
+        {
+            var prop = switchesType.GetProperty(
+                name, BindingFlags.Public | BindingFlags.Static);
+            if (prop == null)
+            {
+                Assert.Fail($"Unable to find {name} property.");
+            }
+            property = prop;
+        }
+
+        // Acquire handles to all of the public properties of
+        // LocalAppContextSwitches.
+        InitProperty(
+            "LegacyRowVersionNullBehavior",
+            out _legacyRowVersionNullBehaviorProperty);
+        InitProperty(
+            "SuppressInsecureTLSWarning",
+            out _suppressInsecureTLSWarningProperty);
+        InitProperty(
+            "MakeReadAsyncBlocking",
+            out _makeReadAsyncBlockingProperty);
+        InitProperty(
+            "UseMinimumLoginTimeout",
+            out _useMinimumLoginTimeoutProperty);
+        InitProperty(
+            "LegacyVarTimeZeroScaleBehaviour",
+            out _legacyVarTimeZeroScaleBehaviourProperty);
+        InitProperty(
+            "UseCompatibilityProcessSni",
+            out _useCompatProcessSniProperty);
+        InitProperty(
+            "UseCompatibilityAsyncBehaviour",
+            out _useCompatAsyncBehaviourProperty);
+#if NETFRAMEWORK
+        InitProperty(
+            "DisableTNIRByDefault",
+            out _disableTNIRByDefaultProperty);
+#endif
+
+        // A local helper to capture the original value of a switch.
+        void InitField(string name, out FieldInfo field, out Tristate value)
+        {
+            var fieldInfo =
```

Comment:
Nit - this could definitely all fit on one line
(and having `= ` with nothing after it is a pet peeve)